backends/magma/ceed-magma.h

```diff
@@ -13,6 +13,34 @@
 #include <ceed/backend.h>
 #include <magma_v2.h>
 
+#define MAGMA_MAXTHREADS_1D 128
+#define MAGMA_MAXTHREADS_2D 128
+#define MAGMA_MAXTHREADS_3D 64
+// Define macro for determining number of threads in y-direction
+// for basis kernels
+#define MAGMA_BASIS_NTCOL(x, maxt) (((maxt) < (x)) ? 1 : ((maxt) / (x)))
+// Define macro for computing the total threads in a block
+// for use with __launch_bounds__()
+#define MAGMA_BASIS_BOUNDS(x, maxt) (x * MAGMA_BASIS_NTCOL(x, maxt))
+
+#ifdef CEED_MAGMA_USE_HIP
+#define MAGMA_RTC_MODULE hipModule_t
+#define MAGMA_RTC_FUNCTION hipFunction_t
+#define MAGMA_RTC_COMPILE CeedCompileHip
+#define MAGMA_RTC_GET_KERNEL CeedGetKernelHip
+#define MAGMA_RTC_RUN_KERNEL CeedRunKernelHip
+#define MAGMA_RTC_RUN_KERNEL_DIM CeedRunKernelDimHip
+#define MAGMA_RTC_RUN_KERNEL_DIM_SH CeedRunKernelDimSharedHip
+#else
+#define MAGMA_RTC_MODULE CUmodule
+#define MAGMA_RTC_FUNCTION CUfunction
+#define MAGMA_RTC_COMPILE CeedCompileCuda
+#define MAGMA_RTC_GET_KERNEL CeedGetKernelCuda
+#define MAGMA_RTC_RUN_KERNEL CeedRunKernelCuda
+#define MAGMA_RTC_RUN_KERNEL_DIM CeedRunKernelDimCuda
+#define MAGMA_RTC_RUN_KERNEL_DIM_SH CeedRunKernelDimSharedCuda
```

Comment:
Minor questions - 

Do we want to prepend `CEED_` here?

For the types, do we want to conditionally typedef `CeedMagmaModule` and `CeedMagmaFunction` as `hipModule_t/CUmodule` and `hipFunction_t/CUfunction`? Might make it easier to see that those are types

Reply:
Would you prefer switching the functions to CeedCamelCase as well?  Or just the types?
I guess I thought it would be good for them to be obviously macros, but we have things in libCEED like `CeedCalloc` which is a macro, so it would be consistent to use that style here.

Reply:
For the types, I was thinking of something like
```C
#ifdef CEED_MAGMA_USE_HIP
typedef CeedMagmaModule hipModule_t;
typedef CeedMagmaFunction hipFunction_t;
#else
typedef CeedMagmaModule CUmodule;
typedef CeedMagmaFunction CUfunction;
#endif
```
so that IDEs know these are types.

I could go either way with the CamelCase for the function names. Whichever you feel will be clearer in 6 months when we've half forgotten the specific details of this PR.

Reply:
I decided to go with a style that matches the other backends better.  Hopefully I don't regret it in 6 months. :)